fsm.go

```diff
@@ -85,6 +105,67 @@ func (r *Raft) runFSM() {
 		lastTerm = req.log.Term
 	}
 
+	commitBatch := func(reqs []*commitTuple) {
+		if !batchingEnabled {
+			for _, ct := range reqs {
+				commitSingle(ct)
+			}
+			return
+		}
+
+		// Only send LogCommand and LogConfiguration log types. LogBarrier types
+		// will not be sent to the FSM.
+		shouldSend := func(l *Log) bool {
+			switch l.Type {
+			case LogCommand, LogConfiguration:
+				return true
+			}
+			return false
+		}
+
+		var lastBatchIndex, lastBatchTerm uint64
+		sendLogs := make([]*Log, 0, len(reqs))
+		for _, req := range reqs {
+			if shouldSend(req.log) {
+				sendLogs = append(sendLogs, req.log)
+			}
+			lastBatchIndex = req.log.Index
+			lastBatchTerm = req.log.Term
+		}
+
+		var responses []interface{}
+		if len(sendLogs) > 0 {
+			start := time.Now()
+			responses = batchingFSM.ApplyBatch(sendLogs)
+			metrics.MeasureSince([]string{"raft", "fsm", "applyBatch"}, start)
+			metrics.SetGauge([]string{"raft", "fsm", "applyBatchNum"}, float32(len(reqs)))
```

Comment:
🤔 what is this useful for? It's a gauge which is a single value not aggregated so each call will "override" the previous until the timeslice is up (e.g. 10s for statsd typically). I feel like if we want to indicate the average/max/min size of batches we'd need a different metric type. In pure statsd the only type that would work would unintuitively be a "timer". Other systems (Prometheus, Dogstatsd) have support for Histogram types which are what you need if you want to show the distribution (avg, max, etc) of sizes of things.

I'd suggest we just leave this out or now as I don't think it will be useful as it is - it's effectively uncontrolled random sampling of the size of the batch at any given metrics interval.